docs/docs/how-tos/recursion-limit.ipynb

```diff
@@ -88,44 +107,44 @@
     "    aggregate: Annotated[list, operator.add]\n",
     "\n",
     "\n",
-    "def node_a(state):\n",
-    "    return {\"aggregate\": [\"I'm A\"]}\n",
+    "def a(state: State):\n",
+    "    print(f'Node A sees {state[\"aggregate\"]}')\n",
+    "    return {\"aggregate\": [\"A\"]}\n",
     "\n",
     "\n",
-    "def node_b(state):\n",
-    "    return {\"aggregate\": [\"I'm B\"]}\n",
+    "def b(state: State):\n",
+    "    print(f'Node B sees {state[\"aggregate\"]}')\n",
+    "    return {\"aggregate\": [\"B\"]}\n",
     "\n",
     "\n",
-    "def node_c(state):\n",
-    "    return {\"aggregate\": [\"I'm C\"]}\n",
+    "# Define nodes\n",
+    "builder = StateGraph(State)\n",
+    "builder.add_node(a)\n",
+    "builder.add_node(b)\n",
     "\n",
     "\n",
-    "def node_d(state):\n",
-    "    return {\"aggregate\": [\"I'm A\"]}\n",
+    "# Define edges\n",
+    "def route(state: State) -> Literal[\"b\", END]:\n",
```

Comment:
i wonder if we could restructure this a bit to present this as another way to control a loop, e.g.:
- control via a conditional edge w/ some termination condition (could also link to react agent docs for this as an example)
- control via recursion limit